perun-core/src/main/java/cz/metacentrum/perun/core/api/exceptions/MfaPrivilegeException.java

```diff
@@ -0,0 +1,32 @@
+package cz.metacentrum.perun.core.api.exceptions;
+
+import cz.metacentrum.perun.core.api.PerunSession;
+import cz.metacentrum.perun.core.api.exceptions.rt.PerunRuntimeException;
+
+/**
+ * This exception is thrown when principal is performing MFA-requiring action but is not authenticated with Multi-Factor
+ * @author Johana Supikova <xsupikov@fi.muni.cz>
+ */
+public class MfaPrivilegeException extends PerunRuntimeException {
```

Comment:
I was thinking about this, and we could make this exception a checked one. Unchecked exceptions usually reflect errors inside the program logic, but this exception represents a common situation. What do you think @Johaney-s @mattjoke @HejdaJakub ?

Reply:
On the other side, we often use authorizing methods in lambda functions in streams, which can't reasonably handle checked exceptions. Hence, if we made this exception a checked one, we would also have to change a lot of code to make it work.

Reply:
From the design standpoint, it would be better, but we have to (as you mentioned) consider the amount of code needed to be updated and re-coded.

Reply:
we decided to keep it a RuntimeException for there are too many stream usages and the exception is not meant to be caught anyway.